src/Chart/renderSegments.tsx

```diff
@@ -52,6 +52,10 @@ export default function renderSegments(
   const paths = data.map((dataEntry, index) => {
     const startAngle = props.startAngle + dataEntry.startOffset;
 
+    if (props.filterSegments({ ...dataEntry }, index) === true) {
+      return null;
+    }
+
```

Comment:
The `if` condition is repeated for each segment. What if we try to take the filtering out from this `.map` loop?

Somewhere around line 52 we might:
1. filter `data` entries only if `props.filterSegments` is defined
2. map the filtered data as usual

Do you see any neat way to do so?

Reply:
Sounds good to me!